### ISSUE TITLE
NOKIA G20 (TA-1336) - ANDROID 11 - 00WW_1_180

### ISSUE THREAD

**Issue by txemasm** [bkerler/mtkclient#202]
Hi,

I've tried to run any command against the mobile with no success. All the time the Client tries to connect but never reaches the phone although phone is on FASTBOOT screen. I've tried on a PC (Windows 10), MacBook (Big Sur) and even the LiveDVD image. Also, I've tried changing the cable but I can access the phone through adb and fastboot so cables seem working fine. Any thought or idea? Thanks a lot!

**Comment by SmallPP420** [comment]
hey this tool DOES NOT communicate with fastboot you have to turn ur phone completely off and connect while holding volume down :)

**Comment by txemasm** [comment]
when I turn on the phone + volume down, only appears the fastboot screen :(

**Comment by SmallPP420** [comment]
dont press power!!!

connect cable while pressing vol down

**Comment by SmallPP420** [comment]
also make sure on the command window its saying

waiting for preloader vcom...

and then connect the cable while pressing vol down

**Comment by txemasm** [comment]
F**k! I don’t know why I tried all the time to turn on the phone!

Anyway, doesn’t work either. Two possibilities:

1) When I connect the cable to the Mac with the volume down pressed, phone tries to turn on and fastboot screen is displayed.

2) If i don’t press the button until android one splash screen or battery charging animation, mtkclient freezes and stop painting dots and the message.

**Comment by SmallPP420** [comment]
do u have mtk drivers installed by any chance and usbdsk driver

**Comment by txemasm** [comment]
Yep. I have both installed. UsbDkController.exe was manually set as environment variable as didn't work properly. I will try to get other drivers and test :)

**Comment by bkerler** [comment]
For Nokia devices you need to press all hw buttons on the phone while connecting usb, otherwise it won't enter the brom. Also I strongly recommend to use linux instead of windows.

**Comment by txemasm** [comment]
Hi guys!

I've been able to unlock the device from the LiveDVD!

The steps that I did were enter on Fastboot mode --> Fastboot Menu --> Enter recovery --> Wipe data/factory reset --> Power off --> Unplug phone --> Plug it pressing Vol up + Vol down.

Problem is that I think I've bricked the phone u_u. When I try to dump the boot and vbmeta it throws errors and when I try to reboot the phone I see the screen about bootloader is unlocked, android one splash screen, and then another screen that says "Your device is corrupt". Always in a loop.

When I try to dump boot and vbmeta this is the message: 

...Port - Device detected :)
Preloader - 	CPU:			MT6765(Helio P35/G35)
Preloader - 	HW version:		0x0
Preloader - 	WDT:			0x10007000
Preloader - 	Uart:			0x11002000
Preloader - 	Brom payload addr:	0x100a00
Preloader - 	DA payload addr:	0x201000
Preloader - 	CQ_DMA addr:		0x10212000
Preloader - 	Var1:			0x25
Preloader - Disabling Watchdog...
Preloader - HW code:			0x766
Preloader - Target config:		0xe7
Preloader - 	SBC enabled:		True
Preloader - 	SLA enabled:		True
Preloader - 	DAA enabled:		True
Preloader - 	SWJTAG enabled:		True
Preloader - 	EPP_PARAM at 0x600 after EMMC_BOOT/SDMMC_BOOT:	False
Preloader - 	Root cert required:	False
Preloader - 	Mem read auth:		True
Preloader - 	Mem write auth:		True
Preloader - 	Cmd 0xC8 blocked:	True
Preloader - Get Target info
Preloader - BROM mode detected.
Preloader - 	HW subcode:		0x8a00
Preloader - 	HW Ver:			0xca00
Preloader - 	SW Ver:			0x0
Preloader - ME_ID:			A49EFDFC0585596E778E7D17417BB9A0
Preloader - SOC_ID:			EAE6A0E43DDEEBBBE3EE2CAB7371835C86D83B403A5F57A1E694329766F70112
PLTools - Loading payload from mt6765_payload.bin, 0x264 bytes
PLTools - Kamakiri / DA Run
Kamakiri - Trying kamakiri2..
Kamakiri - Done sending payload...
PLTools - Successfully sent payload: /opt/mtkclient/mtkclient/payloads/mt6765_payload.bin
Port - Device detected :)
DA_handler - Device is protected.
DA_handler - Device is in BROM mode. Trying to dump preloader.
DAXFlash - Uploading xflash stage 1 from MTK_AllInOne_DA_5.2136.bin
DAXFlash - Successfully uploaded stage 1, jumping ..
Preloader - Jumping to 0x200000
Preloader - Jumping to 0x200000: ok.
DAXFlash - Successfully received DA sync
DAXFlash - DRAM config needed for : 880103534c443634
DAXFlash - Sending emi data ...
DAXFlash - Sending emi data succeeded.
DAXFlash - Uploading stage 2...
DAXFlash - Successfully uploaded stage 2
DAXFlash - EMMC FWVer:      0x0
DAXFlash - EMMC ID:         SLD64G
DAXFlash - EMMC CID:        880103534c4436344760ac7c20b158e9
DAXFlash - EMMC Boot1 Size: 0x400000
DAXFlash - EMMC Boot2 Size: 0x400000
DAXFlash - EMMC GP1 Size:   0x0
DAXFlash - EMMC GP2 Size:   0x0
DAXFlash - EMMC GP3 Size:   0x0
DAXFlash - EMMC GP4 Size:   0x0
DAXFlash - EMMC RPMB Size:  0x400000
DAXFlash - EMMC USER Size:  0xe67800000
DAXFlash - DA-CODE      : 0x666D0
DAXFlash - DA Extensions successfully added
DA_handler - Requesting available partitions ....
DA_handler
DA_handler - [LIB]: Error: Couldn't detect partition: boot
Available partitions:
DA_handler - boot_para
DA_handler - para
DA_handler - expdb
DA_handler - frp
DA_handler - nvcfg
DA_handler - nvdata
DA_handler - md_udc
DA_handler - metadata
DA_handler - protect1
DA_handler - protect2
DA_handler - seccfg
DA_handler - persist
DA_handler - sec1
DA_handler - proinfo
DA_handler - efuse
DA_handler - nvram
DA_handler - wt_tag
DA_handler - logo_a
DA_handler - md1img_a
DA_handler - spmfw_a
DA_handler - scp_a
DA_handler - sspm_a
DA_handler - gz_a
DA_handler - lk_a
DA_handler - boot_a
DA_handler - vendor_boot_a
DA_handler - dtbo_a
DA_handler - tee_a
DA_handler - vbmeta_a
DA_handler - vbmeta_system_a
DA_handler - vbmeta_vendor_a
DA_handler - picture_a
DA_handler - wt_custom_a
DA_handler - logo_b
DA_handler - md1img_b
DA_handler - spmfw_b
DA_handler - scp_b
DA_handler - sspm_b
DA_handler - gz_b
DA_handler - lk_b
DA_handler - boot_b
DA_handler - vendor_boot_b
DA_handler - dtbo_b
DA_handler - tee_b
DA_handler - vbmeta_b
DA_handler - vbmeta_system_b
DA_handler - vbmeta_vendor_b
DA_handler - picture_b
DA_handler - wt_custom_b
DA_handler - oem
DA_handler - elabel
DA_handler - super
DA_handler - userdata
DA_handler - otp
DA_handler - flashinfo
DA_handler
DA_handler - [LIB]: Error: Couldn't detect partition: vbmeta
Available partitions:
DA_handler - boot_para
DA_handler - para
DA_handler - expdb
DA_handler - frp
DA_handler - nvcfg
DA_handler - nvdata
DA_handler - md_udc
DA_handler - metadata
DA_handler - protect1
DA_handler - protect2
DA_handler - seccfg
DA_handler - persist
DA_handler - sec1
DA_handler - proinfo
DA_handler - efuse
DA_handler - nvram
DA_handler - wt_tag
DA_handler - logo_a
DA_handler - md1img_a
DA_handler - spmfw_a
DA_handler - scp_a
DA_handler - sspm_a
DA_handler - gz_a
DA_handler - lk_a
DA_handler - boot_a
DA_handler - vendor_boot_a
DA_handler - dtbo_a
DA_handler - tee_a
DA_handler - vbmeta_a
DA_handler - vbmeta_system_a
DA_handler - vbmeta_vendor_a
DA_handler - picture_a
DA_handler - wt_custom_a
DA_handler - logo_b
DA_handler - md1img_b
DA_handler - spmfw_b
DA_handler - scp_b
DA_handler - sspm_b
DA_handler - gz_b
DA_handler - lk_b
DA_handler - boot_b
DA_handler - vendor_boot_b
DA_handler - dtbo_b
DA_handler - tee_b
DA_handler - vbmeta_b
DA_handler - vbmeta_system_b
DA_handler - vbmeta_vendor_b
DA_handler - picture_b
DA_handler - wt_custom_b
DA_handler - oem
DA_handler - elabel
DA_handler - super
DA_handler - userdata
DA_handler - otp
DA_handler - flashinfo

**Comment by SmallPP420** [comment]
try flashing the empty vbmeta by

python mtk w vbmeta_a vbmeta.img.empty
python mtk w vbmeta_b vbmeta.img.empty

**Comment by txemasm** [comment]
Hi,

Finally I flashed an stock firmware through ‘adb sideload’. Now phone boots and its bootloader is unlocked. Next step will be rooting it. I think that this thread can be closed.

Thank you very much indeed for making this awesome tool and for helping me with my devices. You rock!

**Comment by noomC** [comment]
Hi,
I've same issue but with G20 (TA-1336) ver. 00WW_1_190, I tried PC (Windows 10) and the LiveDVD but phone doesn't enter brom: phone turned off then press Vol up + Vol down and insert cable but nothing shows up in device manager/mtk_gui.



> Fastboot mode --> Fastboot Menu --> Enter recovery --> Wipe data/factory reset --> Power off --> Unplug phone --> Plug it pressing Vol up + Vol down.

I tried to follow your steps but when I plug the cable no device is detected, I was able to enter recovery and perform data/factory reset with the following commands:
`fastboot reboot fastboot` 
`fastboot reboot recovery`



> For Nokia devices you need to press all hw buttons on the phone while connecting usb

I also tried every possible combination of the 4 HW buttons (Vol+, Vol-, Power, Google Assistant/Google Search key).

I know that COM device should appear almost instantly after cable plugged in, some time ago I flashed a Lenovo with MTK processor, the Nokia device should vibrate as it does at startup when goes in brom?

Any idea or advice?
Thanks
